torchao/dtypes/floatx/float8_layout.py

```diff
@@ -311,3 +311,80 @@ def _linear_fp_act_fp8_weight_impl(
     bias: Optional[torch.Tensor],
 ):
     return torch.nn.functional.linear(input_tensor, weight_tensor.dequantize(), bias)
+
+
+def _sdpa_float8_check(
+    q: Union[torch.Tensor, "AffineQuantizedTensor"],
+    k: Union[torch.Tensor, "AffineQuantizedTensor"],
+    v: Union[torch.Tensor, "AffineQuantizedTensor"],
+    kwargs,
+) -> bool:
+    def is_compatible_per_tensor_float8_aqt(t):
+        # tensor is float8 quantized affine quantized tensor
+        return (
+            isinstance(t, AffineQuantizedTensor)
+            and isinstance(t._layout, Float8Layout)
+            and t.tensor_impl.dtype in [torch.float8_e4m3fn, torch.float8_e5m2]
+            and (t.shape == t.block_size)
+            and t.shape[-1] in [64, 128, 256]
+        )
+
+    dropout_p = kwargs.get("dropout_p", 0.0)
+
+    return (
+        is_compatible_per_tensor_float8_aqt(q)
+        and is_compatible_per_tensor_float8_aqt(k)
+        and is_compatible_per_tensor_float8_aqt(v)
+        and "attn_mask" not in kwargs
+        and dropout_p == 0.0
+    )
+
+
+def _sdpa_float8_impl(
+    q: Union[torch.Tensor, "AffineQuantizedTensor"],
+    k: Union[torch.Tensor, "AffineQuantizedTensor"],
+    v: Union[torch.Tensor, "AffineQuantizedTensor"],
+    kwargs,
+) -> torch.Tensor:
+    # requires build from source
+    # https://github.com/Dao-AILab/flash-attention/tree/main?tab=readme-ov-file#flashattention-3-beta-release
+    # for libc10.so
+    import torch
+    from hopper.flash_attn_interface import flash_attn_func
+
+    q_tensor_impl = q.tensor_impl
+    assert not q_tensor_impl.transposed
+    q_float8_data = q_tensor_impl.float8_data
+    # change from scalar to tensor of size [1]
+    q_scale = q_tensor_impl.scale
+    q_scale = torch.tensor([q_scale], device=q_scale.device)
```

Comment:
are the scales on host?

Reply:
you mean `q_scale` before we call `torch.tensor`? they should be using the same device as original weight I think, so should be on cuda before